pkg/virt-controller/watch/migration.go

```diff
@@ -522,6 +522,19 @@ func (c *MigrationController) sync(key string, migration *virtv1.VirtualMachineI
 			// migration was accepted into the system, now see if we
 			// should create the target pod
 			if vmi.IsRunning() {
+				vmiCopy := vmi.DeepCopy()
```

Comment:
The place for the fix is right. I think we should work a little on the idea on how to fix this: We already have pod expectations set. This means that the controller loop will not proceed until it sees the created pod in its cache. The fix here only reduced the chance that multiple pods get created. Since there are no transactions, if the controller fails to update the status, it would still create pods over and over.

What we are however missing, is a check in the pod cache to see if the pod already exists, before just blindly creating another one.

Reply:
Did you notice the whole function is under a lock?
Can't we rely on VMI MigrationState instead of pod creation?

Reply:
> Did you notice the whole function is under a lock?

The lock does not make any difference. You are not racing between different create-calls in parallel. There is anyway already a 

> Can't we rely on VMI MigrationState instead of pod creation?

You can, but then you would have to first update the VMI status and THEN create the pod. In case the pod-creation goes wrong you will have to cope with the outdated VMI status on the next controller invocation.